apps/browser-extension-wallet/src/lib/scripts/migrations/versions/v1_8_2.ts

```diff
@@ -0,0 +1,130 @@
+      return {
+        type: WalletType.InMemory,
+        extendedAccountPublicKey: data.extendedAccountPublicKey,
+        encryptedSecrets: {
+          rootPrivateKeyBytes: HexBlob.fromBytes(Buffer.from(data.encryptedRootPrivateKeyBytes)),
+          keyMaterial: HexBlob.fromBytes(Buffer.from(JSON.parse(mnemonic).data))
+        },
+        metadata: {
+          name: getWalletFromStorage()?.name
+        }
+      };
+    }
+    case Wallet.KeyManagement.KeyAgentType.Ledger:
+      return {
+        type: WalletType.Ledger,
+        extendedAccountPublicKey: data.extendedAccountPublicKey
+      };
+    case Wallet.KeyManagement.KeyAgentType.Trezor:
+      return {
+        type: WalletType.Trezor,
+        extendedAccountPublicKey: data.extendedAccountPublicKey
+      };
+    default:
+      throw new Error('Unknown key agent type');
+  }
+};
+
+export const v_1_8_2: Migration = {
+  version: MIGRATION_VERSION,
+  requiresPassword: () => {
+    const lock = getItemFromLocalStorage<any>(LOCK_STORAGE);
+    const keyAgentData = getItemFromLocalStorage<any>(KEY_AGENT_DATA_STORAGE);
+
+    if (isWalletLoacked(lock, keyAgentData)) {
+      return true;
+    }
+
+    return false;
+  },
+  upgrade: async (password) => {
+    const lock = getItemFromLocalStorage<any>(LOCK_STORAGE);
+    const keyAgentData =
+      getItemFromLocalStorage<any>(KEY_AGENT_DATA_STORAGE) || (await decryptLock(lock.data, password)).keyAgentData;
+
+    return {
+      prepare: () => void 0,
+      assert: () => void 0,
+      persist: async () => {
+        console.info(`Persisting migrated data for ${MIGRATION_VERSION} upgrade`);
+
+        const wallets = await firstValueFrom(walletRepository.wallets$);
+        const walletProps = await keyAgentDataToAddWalletProps(keyAgentData);
+        const walletId = await getWalletId(
+          walletProps.type === WalletType.Script ? walletProps.script : walletProps.extendedAccountPublicKey
+        );
+        const lockValue =
+          keyAgentData.__typename === Wallet.KeyManagement.KeyAgentType.InMemory ? HexBlob.fromBytes(lock) : undefined;
+
+        if (!wallets.some((wallet) => wallet.walletId === walletId)) {
+          await walletRepository.addWallet(await keyAgentDataToAddWalletProps(keyAgentData));
+
+          await walletRepository.addAccount({
+            accountIndex: keyAgentData.accountIndex,
+            metadata: { name: 'Account #0', lockValue },
+            walletId
+          });
+
+          await walletManager.activate({
+            chainId: keyAgentData.chainId,
+            walletId,
+            accountIndex: keyAgentData.accountIndex,
+            provider
+          });
+        }
+
+        if (isWalletLoacked(lock, keyAgentData)) {
+          setItemInLocalStorage(LOCK_STORAGE, lockValue);
+        } else {
+          removeItemFromLocalStorage(LOCK_STORAGE);
+        }
+
+        removeItemFromLocalStorage(KEY_AGENT_DATA_STORAGE);
```

Comment:
This should also be obsolete
```suggestion
        removeItemFromLocalStorage(KEY_AGENT_DATA_STORAGE);
        removeItemFromLocalStorage('wallet');
```